File: MyDictionaryAndroid/app/src/main/java/com/dor/mydictionary/core/Difficulty.kt
package com.dor.mydictionary.core

import androidx.compose.ui.graphics.Color

enum class Difficulty(val displayName: String, val level: Int, val color: Color) {
    New("New", 0, Color(0xFF9E9E9E)), // Gray
    InProgress("In Progress", 1, Color(0xFFFF9800)), // Orange
    NeedsReview("Needs Review", 2, Color(0xFFF44336)), // Red
    Mastered("Mastered", 3, Color(0xFF4CAF50)); // Green
    
    companion object {
        fun fromLevel(level: Int): Difficulty {
            return values().find { it.level == level } ?: New
        }
        
        fun fromScore(score: Int): Difficulty {
            return when {
                score < 0 -> NeedsReview
                score >= 1 && score <= 49 -> InProgress
                score >= 50 -> Mastered
                else -> New
            }
        }
    }
} 

File: Shared/Core/Models/Difficulty.swift
//
//  Difficulty.swift
//  My Dictionary
//
//  Created by Aleksandr Riakhin on 3/8/25.
//

import Foundation

enum Difficulty: String, CaseIterable {
    case new = "new"
    case inProgress = "inProgress"
    case needsReview = "needsReview"
    case mastered = "mastered"
    
    var displayName: String {
        switch self {
        case .new:
            return "New"
        case .inProgress:
            return "In Progress"
        case .needsReview:
            return "Needs Review"
        case .mastered:
            return "Mastered"
        }
    }
    
    var color: String {
        switch self {
        case .new:
            return "secondary"
        case .inProgress:
            return "orange"
        case .needsReview:
            return "red"
        case .mastered:
            return "green"
        }
    }
    
    var level: Int32 {
        switch self {
        case .new:
            return 0
        case .inProgress:
            return 1
        case .needsReview:
            return 2
        case .mastered:
            return 3
        }
    }
} 